### PR TITLE
ARROW-7066: [Python] Allow returning ChunkedArray in __arrow_array__

### PR DESCRIPTION
**Pull request by jorisvandenbossche**

### CODE REVIEW

File: python/pyarrow/tests/test_array.py [hunk 1]
```diff
@@ -1893,6 +1893,19 @@ def __arrow_array__(self, type=None):
...
+    class MyArray2:
+        def __init__(self, data):
+            self.data = data
+
+        def __arrow_array__(self, type=None):
+            return pa.chunked_array([self.data], type=type)
+
+    arr = MyArray2(np.array([1, 2, 3], dtype='int64'))
+    result = pa.array(arr)
+    expected = pa.array([1, 2, 3], type=pa.int64())
```
Comment (xhochy) [inline python/pyarrow/tests/test_array.py]: Well, this also needs to be a chunked array here.
```suggestion
    expected = pa.chunked_array([[1, 2, 3]], type=pa.int64())
```

Comment (jorisvandenbossche) [reply python/pyarrow/tests/test_array.py]: Whoops, that was indeed what I wanted to test ;)

**Review by pitrou — APPROVED** [review]
+1

**Comment by pitrou** [comment]
This PR needs rebasing for the CI changes.

**Comment by jorisvandenbossche** [comment]
Rebased this, and all green except waiting on Appveyor

**Pull request closed without merging**